projects/buildtarget.vala

```diff
@@ -0,0 +1,98 @@
+using Gee;
+
+/**
+ * Represents a single build target from a build script.
+ * Usage:
+ * 1. configure a build directory for a build script
+ * 2. generate a BuildTarget for each target in the build directory
+ * 3. generate a Compilation for each target source in each BuildTarget
```

Comment:
Does this mean that we have a code context for each source file? Doesn't that use a lot of memory & is slower?

Reply:
No. We have a code context for each Meson target. You'll see what I mean if you clone a project like Geary or Granite and run `meson build && meson introspect --targets | python -m json.tool | gedit -`